src/Microsoft.Graph.Core/Requests/RedirectHandler.cs

```diff
@@ -0,0 +1,159 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Graph
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Text;
+    using System.Net.Http.Headers;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using System.Net;
+
+    /// <summary>
+    /// An <see cref="DelegatingHandler"/> implementation using standard .NET libraries.
+    /// </summary>
+    public class RedirectHandler : DelegatingHandler
+    {
+
+        private const int maxRedirects = 5;
+
+        /// <summary>
+        /// Constructs a new <see cref="RedirectHandler"/> 
+        /// </summary>
+        /// <param name="innerHandler">An HTTP message handler to pass to the <see cref="HttpMessageHandler"/> for sending requests.</param>
+        public RedirectHandler(HttpMessageHandler innerHandler)
+        {
+            InnerHandler = innerHandler;
+        }
+
+        /// <summary>
+        /// Disposes the HttpMessageHandler instance.
+        /// </summary>
+        public new void Dispose()
+        {
+            if (InnerHandler != null)
+            {
+                InnerHandler.Dispose();
+            }
+
+        }
+
+        /// <summary>
+        /// Sends the Request 
+        /// </summary>
+        /// <param name="request">The <see cref="HttpRequestMessage"/> to send.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/>for the request.</param>
+        /// <returns>The <see cref="HttpResponseMessage"/>.</returns>
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            
+            // send request first time to get response
+            var response = await base.SendAsync(request, cancellationToken);
+
+            // check response status code 
+            if (IsRedirect(response.StatusCode))
+            {
+
+                // general copy request with internal copyRequest(see copyRequest for details) Method 
+                var newRequest = CopyRequest(request);
+                StreamContent content = null;
+
+                if (request.Content != null && request.Content.Headers.ContentLength != 0)
+                {
+                    content = new StreamContent(request.Content.ReadAsStreamAsync().Result);
```

Comment:
We cannot use .Result here as it will break in many environments.  We need to use await.  Just curious, is there a reason that this copying is not done in CopyRequest?

Reply:
Yes. You are right. Thanks for suggestions. I planned to skip copy content each time when re-issued a request, but I have ignored the important thing that re-issue a request should be based on previous request and response, not the original request. Will update them. Thanks.

Reply:
Btw, I'll update methods and re-test them, whether it's ok to close this one and create a new PR with updates?

Reply:
You can just update this PR.